jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -353,36 +317,67 @@ public String toString()
         return String.format("%s@%x", this.getClass().getSimpleName(), hashCode());
     }
 
-    /**
-     * Simple bundle of information that is cached in the SSLSession. Stores the
-     * effective keySize and the client certificate chain.
-     */
-    private static class CachedInfo
+    private class SslAttributes extends Attributes.Wrapper
     {
-        private final X509Certificate[] _certs;
-        private final Integer _keySize;
-        private final String _idStr;
-
-        CachedInfo(Integer keySize, X509Certificate[] certs, String idStr)
-        {
-            this._keySize = keySize;
-            this._certs = certs;
-            this._idStr = idStr;
-        }
+        private final Request _request;
+        private final SSLSession _session;
 
-        X509Certificate[] getCerts()
+        public SslAttributes(Request request, SSLSession sslSession, Attributes attributes)
```

Comment:
This class now has a weird behaviour that if you set one of these `JAVAX_SERVLET_REQUEST_*` attributes, it sets it on the wrapped attributes, but when accessing the attributes you only get the version stored here. Maybe you should override `setAttribute` as well.

Reply:
yeah probably.  Just not clear what it should do?  Replace the passed value?
Maybe in get, we should try the wrapped one first and then only use the special handling if it is null.   Yeah will do it that way....

Reply:
So what happens now if an `SslAttributes` was wrapping another `Attributes` with one of these attributes already set, you would want `SslAttributes` to override the behaviour of the wrapped attributes and not the other way around.

I feel like it would be a better fix to store the SSL attributes at this level only, similar to the pattern used in `ForwardAttributes` and `IncludeAttributes`. Seems weird that they would be stored both here and in the wrapped attributes.

But I'll just approve and leave it up to you, PR looks good to me in general.